src/app/tasks/task.service.ts

```diff
@@ -40,59 +42,69 @@ export class TaskService {
     }
 
     private convertBasicTaskToTask(basicTask: BasicTask) : Task | null {
-        let taskIndex: number = this.tasksSubject.getValue().findIndex(task => task.id == basicTask.id);
+        let taskIndex: number = this.tasks.findIndex(task => task.id == basicTask.id);
         if(taskIndex === -1) return null;
-        return this.tasksSubject.getValue()[taskIndex];
+        return this.tasks[taskIndex];
     }
 
     private convertTasksToBasicTasks(tasks: Task[]) : BasicTask[] {
         let basicTasks : BasicTask[] = [];
-        this.tasksSubject.getValue().forEach(task => basicTasks.push(this.convertTaskToBasicTask(task)));
+        this.tasks.forEach(task => basicTasks.push(this.convertTaskToBasicTask(task)));
         return basicTasks;
     }
 
     public getTasks() : BasicTask[] {
-        return this.convertTasksToBasicTasks(this.tasksSubject.getValue()).slice()
+        return this.convertTasksToBasicTasks(this.tasks).slice()
         .sort(
             (taskA, taskB) => {
                 return (taskA.title < taskB.title) ? -1 : (taskA.title > taskB.title) ? 1 : 0
         });
     }
 
-    public getTask(taskId: number): Task | null {
-        return this.tasksSubject.getValue()?.find(task => {
-            return task.id == taskId;
-        }) ?? null;
+    public getTask(taskId: number): Observable<Task | null> {
+        return this.httpClient.get<Task | null>(beApiRoutes.taskDetails + taskId);
     }
 
     public addTask(task: Task) {
-        task.id = this.idGeneratorService.generateId();
-        this.tasksSubject.getValue().push(task);
+        let oldTaskList = this.tasks.slice();
+        this.httpClient.post<Task | null>(beApiRoutes.tasks, task)
+        .subscribe(createdTask => {
+            if(createdTask) {
+                oldTaskList.push(createdTask);
+                this.tasks = oldTaskList;
+                this.basicTasksSubject.next(this.getTasks());
+            }
+        });
     }
 
-    public updateTask(id: number, editedTask: Task){
+    public updateTask(taskId: number, editedTask: Task){
         // Update the task if the ids match
-        this.tasksSubject.next(this.tasksSubject.getValue().map((task: Task) => {
-            if(task.id === id) task = editedTask;
-            return task
-        }));
+        this.httpClient.patch<Task | null>(beApiRoutes.taskDetails + taskId, editedTask)
+        .subscribe(updatedTask => {
```

Comment:
same here don't subscribe in service, instead you can call subscribe on the component level